### PR TITLE
[GEOS-9956] Clean up 'Class is a raw type' compile warnings

### PR DESCRIPTION
**Pull request by groldan**
- 448 warnings in core modules
- 35 in extensions
- 105 in "community release" modules

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.**

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [ ] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by Continuous Integration after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by Continuous Integration after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

### CODE REVIEW

File: src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/processor/CatalogItemProcessor.java [hunk 1]
```diff
@@ -198,7 +198,9 @@ public T process(T resource) throws Exception {
                                     .getResourceByName(((ResourceInfo) resource).getName(), clz);
                     if (source != null && source.getStore() != null) {
                         StoreInfo ds =
-                                getCatalog().getStoreByName(source.getStore().getName(), clz);
+                                getCatalog()
```
Comment (aaime) [inline src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/processor/CatalogItemProcessor.java:201]: This one gave me pause, what's happening here? The old code did not make much sense TBH, but I would have expected to see CoveageStoreInfo for coverages and DataStoreInfo for rasters. Then again, I believe one cannot give the same name to two stores, so it's probably fine.

File: src/community/geopkg/src/main/java/org/geoserver/geopkg/wps/GeneralizedTablesExtension.java [hunk 2]
```diff
@@ -51,7 +51,8 @@ public GeoPkgExtension getExtension(String name, GeoPackage geoPackage) {
         }
 
         @Override
-        public GeoPkgExtension getExtension(Class extensionClass, GeoPackage geoPackage) {
+        public GeoPkgExtension getExtension(
+                @SuppressWarnings("rawtypes") Class extensionClass, GeoPackage geoPackage) {
```
Comment (aaime) [inline src/community/geopkg/src/main/java/org/geoserver/geopkg/wps/GeneralizedTablesExtension.java:55]: Wondering if GeoPkgExtensionFactory should be fixed instead. Even just setting a Class<?> there.

File: src/community/geopkg/src/main/java/org/geoserver/geopkg/wps/PortrayalExtension.java [hunk 3]
```diff
@@ -59,7 +59,8 @@ public GeoPkgExtension getExtension(String name, GeoPackage geoPackage) {
         }
 
         @Override
-        public GeoPkgExtension getExtension(Class extensionClass, GeoPackage geoPackage) {
+        public GeoPkgExtension getExtension(
```
Comment (aaime) [inline src/community/geopkg/src/main/java/org/geoserver/geopkg/wps/PortrayalExtension.java:62]: Same as above

File: src/community/geopkg/src/main/java/org/geoserver/geopkg/wps/SemanticAnnotationsExtension.java [hunk 4]
```diff
@@ -50,7 +50,8 @@ public GeoPkgExtension getExtension(String name, GeoPackage geoPackage) {
         }
 
         @Override
-        public GeoPkgExtension getExtension(Class extensionClass, GeoPackage geoPackage) {
+        public GeoPkgExtension getExtension(
+                @SuppressWarnings("rawtypes") Class extensionClass, GeoPackage geoPackage) {
```
Comment (aaime) [inline src/community/geopkg/src/main/java/org/geoserver/geopkg/wps/SemanticAnnotationsExtension.java:54]: Same as above

File: src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/ogcapi/OpenAPIBuilder.java [hunk 5]
```diff
@@ -81,7 +82,7 @@ protected void addBasePathFormats(OpenAPI api) {
      * Returns the landing page document class. By default it returns {@link
      * AbstractLandingPageDocument} in case a service has more representations than usual
      */
-    protected Class getLandingPageDocumentClass() {
+    protected Class<AbstractLandingPageDocument> getLandingPageDocumentClass() {
```
Comment (aaime) [inline src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/ogcapi/OpenAPIBuilder.java:85]: Seems like a subclass could return a subclass though. Shoud it be ``<? extends AbstractLandingPageDocument>``?

**Review by aaime — COMMENTED** [review]
Looks good, I've pin-pointed a few cases that made me wonder.

**Comment by aaime** [comment]
@groldan I see you made further work, but saw no response to comments above. I'm guessing you handled conflicts but not yet the review feedback, but... well... just guessing :-D

**Comment by groldan** [comment]
> @groldan I see you made further work, but saw no response to comments above. I'm guessing you handled conflicts but not yet the review feedback, but... well... just guessing :-D

Hey Andrea, what happens is that I didn't want to reply just with a "not the scope of this pr" for the CatalogItemProcessor stuff, and took it for a ride on a separate branch, then waited to have spare time to finish it up. See #4893

File: src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/processor/CatalogItemProcessor.java (hunk 1 above)
Comment (groldan) [reply src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/processor/CatalogItemProcessor.java:201]: Indeed that looks like a bug, although for the sake of this PR it'd be out of scope. I did fix it in a separate PR though, see #4893. A `ResourceInfo` subclass was being used to ask for a `StoreInfo`. That method suffered a lot more in terms of readability, so I tried to make it better in that pr.

File: src/community/geopkg/src/main/java/org/geoserver/geopkg/wps/GeneralizedTablesExtension.java (hunk 2 above)
Comment (groldan) [reply src/community/geopkg/src/main/java/org/geoserver/geopkg/wps/GeneralizedTablesExtension.java:55]: At `GeoPkgExtensionFactory`:
```
 public GeoPkgExtension getExtension(String name, GeoPackage geoPackage);
```
So it's ok to return `GeoPkgExtension` subclassess. Changed this one to return `GeneralizedTablesExtension`

File: src/community/geopkg/src/main/java/org/geoserver/geopkg/wps/PortrayalExtension.java (hunk 3 above)
Comment (groldan) [reply src/community/geopkg/src/main/java/org/geoserver/geopkg/wps/PortrayalExtension.java:62]: Amended to return `PortrayalExtension`

File: src/community/geopkg/src/main/java/org/geoserver/geopkg/wps/SemanticAnnotationsExtension.java (hunk 4 above)
Comment (groldan) [reply src/community/geopkg/src/main/java/org/geoserver/geopkg/wps/SemanticAnnotationsExtension.java:54]: Amended to return `SemanticAnnotationsExtension`

File: src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/ogcapi/OpenAPIBuilder.java (hunk 5 above)
Comment (groldan) [reply src/community/ogcapi/ogcapi-core/src/main/java/org/geoserver/ogcapi/OpenAPIBuilder.java]: Indeed, amended to return `<? extends AbstractLandingPageDocument>`

**Review by aaime — APPROVED** [review]
Ok thanks, looking good, and merging.

**Pull request merged**